plugins/search-backend-module-pg/src/PgSearchEngine/PgSearchEngine.ts

```diff
@@ -26,28 +26,57 @@ import {
   DatabaseStore,
   PgSearchQuery,
 } from '../database';
+import { v4 as uuid } from 'uuid';
+import { Config } from '@backstage/config';
+import { PgSearchHighlightConfig, PgSearchHighlightOptions } from '../types';
 
 export type ConcretePgSearchQuery = {
   pgQuery: PgSearchQuery;
   pageSize: number;
 };
 
 export class PgSearchEngine implements SearchEngine {
-  constructor(private readonly databaseStore: DatabaseStore) {}
+  private readonly highlightOptions: PgSearchHighlightConfig;
+
+  constructor(
+    private readonly databaseStore: DatabaseStore,
+    highlightOptions?: PgSearchHighlightOptions,
+  ) {
+    const uuidTag = uuid();
+    this.highlightOptions = {
+      preTag: `<${uuidTag}>`,
+      postTag: `</${uuidTag}>`,
+      useHighlight: true,
+      maxWords: 35,
+      minWords: 15,
+      shortWord: 3,
+      highlightAll: false,
+      maxFragments: 0,
+      fragmentDelimiter: ' ... ',
+      ...highlightOptions,
+    };
+  }
 
   static async from(options: {
     database: PluginDatabaseManager;
+    config: Config;
   }): Promise<PgSearchEngine> {
     return new PgSearchEngine(
       await DatabaseDocumentStore.create(await options.database.getClient()),
+      options.config.getOptional<PgSearchHighlightOptions>(
+        'search.pg.highlightOptions',
+      ),
     );
   }
 
   static async supported(database: PluginDatabaseManager): Promise<boolean> {
     return await DatabaseDocumentStore.supported(await database.getClient());
   }
 
-  translator(query: SearchQuery): ConcretePgSearchQuery {
+  translator(
+    query: SearchQuery,
+    options: PgSearchHighlightConfig,
```

Comment:
Might be worth typing this to a more [generic options config](https://github.com/backstage/backstage/blob/d9f86c2d81d85975d89729df905c2f4c26736060/plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts#L84-L86) that wraps `PgSearchHighlightConfig` so it can be more easily extended in the future if necessary

Reply:
Just got done with @freben's review, will take a look at doing this now

Reply:
Cleaned up the types while I got this in place, looks better now, thanks again for reviewing this